File: Code/matrix/saddle_point_of_matrix/saddle_point.c
// C program to find the saddle point coordinates in the given matrix
#include <stdio.h>
int main()
{
	int i, j, k, m, n, min, max,pos[2][2];
	/* Input the size of the matrix from the user */
	printf("Enter the square matrix order : ");
	scanf("%d", &m);
	int matrix[m][m];
	/* Input the matrix */
	printf("\nInput the matrix : ");
	for (i = 0; i < m; i++)
	{
		for (j = 0; j < m; j++)
		{
			scanf("%d", &matrix[i][j]);
		}
	}
	printf("\nThe matrix is \n");
	for (i = 0; i < m; i++)
		{
			for (j = 0; j < m; j++)
				{
					printf("%d ", matrix[i][j]);
				}
			printf("\n");
		}
/* find the saddle points in the given matrix */
	for (i = 0; i < m; i++)
		{
			min = matrix[i][0];
			for (j = 0; j < m; j++)
				{
					if (min >= matrix[i][j])
						{
							min = matrix[i][j];
							pos[0][0] = i;
							pos[0][1] = j;
						}
				}
     j = pos[0][1];
     max = matrix[0][j];
	for (k = 0; k < m; k++)
	{
		if (max <= matrix[k][j])
			{
				max = matrix[i][j];
				pos[1][0] = k;
				pos[1][1] = j;
			}
	}
/* saddle point is the minimum of a row and maximum of the column */
	if (min == max) 
    {
		if (pos[0][0] == pos[1][0] &&pos[0][1] == pos[1][1])
		{
			printf("\nSaddle point (%d, %d) : %d",pos[0][0], pos[0][1], max);
		}
	}
}
	return 0;
}

File: Code/matrix/saddle_point_of_matrix/SaddlePoint.java
class SaddlePoint {
	public static void main(String[] args) {
		int matrix[][]= {{1, 2, 3},
						 {4, 5, 6},
						 {7, 8, 9}};
		int order = 3;
		if(findSaddlePoint(matrix,order) == false)
			System.out.println("No Saddle Point");
	}

	private static boolean findSaddlePoint(int[][] a, int n) {
		
		int k;
		// Process all rows one by one
		for(int i=0; i<n; i++) {
			int min_row = a[i][0], col_ind = 0;
			
			//Find the minimum row element and corresponding column
			for(int j=0; j<n; j++) {
				if(min_row > a[i][j]) {
					min_row = a[i][j];
					col_ind = j;
				}
			}
			
			// Check if the minimum element of row is also the maximum element of column or not
			for(k=0; k<n; k++) {
				if(min_row < a[k][col_ind])
					break;
				}
			
			// If saddle point is present in this row then print it
			if(k == n) {
				System.out.println("Value of Saddle Point " + min_row);
				return true;
			}
		}
		
		// If Saddle Point not found
		return false;
	}
}